### ISSUE TITLE
Using the Gherkin keywords Scenario Template in place of Scenario Outline causes scenarios to be executed twice

### ISSUE THREAD

**Issue by fourpastmidnight** [pester/Pester#1123]
## 1. General summary of the issue

Using the synonymous Gherkin keywords `Scenario Template` in place of `Scenario Outline' may cause the scenario associated with those keywords (or the scenario before it) to be executed twice.

## 2. Describe Your Environment

Pester version : 4.4.2 C:\Program Files\WindowsPowerShell\Modules\Pester\4.4.2\Pester.psd1
PowerShell version : 5.1.17134.165
OS version : Microsoft Windows NT 10.0.17134.0

## 3. Expected Behavior

I expected the scenario associated with the keywords (or the scenario preceding the scenario using the keywords) to be executed only once.

## 4.Current Behavior

If you have the following Gherkin file:

```gherkin
Scenario Outline: This scenario will be executed twice

    # A scenario, here
    # With some examples here

@ignore
Scenario Template: This is an ignored scenario

    # Another scenario here
    # More examples here
```

**or** this Gherkin file:

```gherkin
Scenario Template: This scenario will be executed twice

    # A scenario here
    # With some examples here
```

and run Pester with the following command:

    Invoke-Gherkin -Path .\features\Path-to-feature-file.feature -ExcludeTag ignore

then the scenarios indicated in the gherkin files above will be executed twice.

## 5. Possible Solution

If Pester is not already using the latest Gherkin parser (it probably isn't, as other newer Gherkin keywords are not yet supported) then update Pester to use the latest Gherkin parser. Or, for the version of the Gherkin parser in use, ensure that all Gherkin keywords supported by the parser are also supported by Pester.

## 6. Context

I was using the keywords `Scenario Template` and `Scenarios` instead of the older `Scenario Outline` and `Examples` keywords. Note, however, that Pester does not seem to have any issues with interchanging the `Examples` and `Scenarios` keywords.

I had to revert my feature files to use only the `Scenario Outline` keywords.

**Comment by fourpastmidnight** [comment]
@Jaykul Sounds good. As I said in another issue, I've opened up so many in the past two weeks, it's high-time I consider trying to tackle some of these myself! I've just been trying to get a module off the ground for our DevOps processes--so I haven't had a lot of free time.

However, this project has been helping me get my powershell code under test--and it's really well done. I've really enjoyed using the framework. So thanks for all the hard work. I hope I can help contribute soon to help move Pester along.

**Comment by fourpastmidnight** [comment]
@Jaykul checkout the PR I submitted to add in this support.

I also found that Gherkin implemented the `Example` keyword as a synonym for `Scenario`, but that requires v6 of Gherkin, which, if I'm reading their repo correctly, hasn't been officially released yet. (I also looked at NuGet.org, and the latest version is 5.1.0--which Pester may be at 5.0.0, but that's another matter.)
